cmd/common.go

```diff
@@ -237,6 +238,64 @@ func loadConfig(nodetype check.NodeType) string {
 	return filepath.Join(path, file)
 }
 
+func mapToCISVersion(kubeToCISMap map[string]string, kv string) string {
+	cisVersion, found := kubeToCISMap[kv]
+	for !found && (kv != defaultKubeVersion && !isEmpty(kv)) {
+		kv = decrementVersion(kv)
+		cisVersion, found = kubeToCISMap[kv]
+		glog.V(2).Info(fmt.Sprintf("mapToCISVersion for cisVersion: %q found: %t\n", cisVersion, found))
+	}
+
+	if !found {
+		glog.V(2).Info(fmt.Sprintf("mapToCISVersion unable to find a match for: %q", kv))
+		glog.V(2).Info(fmt.Sprintf("mapToCISVersion kubeToCISMap: %#v", kubeToCISMap))
+		return ""
+	}
+
+	return cisVersion
+}
+
+func loadVersionMapping(v *viper.Viper) (map[string]string, error) {
+	versionMappingConf := v.Sub(string(versionMapping))
+	if versionMappingConf == nil {
+		return nil, fmt.Errorf("config file is missing %s section", versionMapping)
+	}
+	kubeToCISMap := versionMappingConf.GetStringMapString(kubersionCISMapping)
+	if kubeToCISMap == nil || (len(kubeToCISMap) == 0) {
+		return nil, fmt.Errorf("config file is missing %s.%s section", versionMapping, kubersionCISMapping)
+	}
+
+	return kubeToCISMap, nil
+}
+
+func getBenchmarkVersion(kubeVersion, benchmarkVersion string, v *viper.Viper) (bv string, err error) {
+	if !isEmpty(kubeVersion) && !isEmpty(benchmarkVersion) {
+		return "", fmt.Errorf("It is an error to specify both --version and --benchmark flags")
+	}
+
+	if isEmpty(benchmarkVersion) {
+		if isEmpty(kubeVersion) {
+			kubeVersion, err = getKubeVersion()
+			if err != nil {
+				return "", fmt.Errorf("Version check failed: %s\nAlternatively, you can specify the version with --version", err)
+			}
+		}
+
+		kubeToCISMap, err := loadVersionMapping(v)
+		if err != nil {
+			return "", err
+		}
+
+		benchmarkVersion = mapToCISVersion(kubeToCISMap, kubeVersion)
+		if isEmpty(benchmarkVersion) {
+			// unable to find a match for kubeVersion
+			return "", fmt.Errorf("Unable to find a matching CIS Version match for kubernetes version: %s", kubeVersion)
+		}
+		glog.V(2).Info(fmt.Sprintf("Mapped Kubernetes version: %s to CIS version: %s", kubeVersion, benchmarkVersion))
```

Comment:
```suggestion
		glog.V(2).Info(fmt.Sprintf("Mapped Kubernetes version: %s to benchmark: %s", kubeVersion, benchmarkVersion))
```

to allow for the possibility of benchmarks that aren't CIS, like Red Hat